### PR TITLE
string_view conversions

### PR DESCRIPTION
**Pull request by neheb**
Signed-off-by: Rosen Penev <rosenp@gmail.com>

### CODE REVIEW

File: src/content/scripting/script.cc [hunk 1]
```diff
@@ -803,7 +803,7 @@ std::string Script::convertToCharset(const std::string& str, charset_convert_t c
         return _f2i->convert(str);
     case J2I:
         return _j2i->convert(str);
-    default:
+    case I2I:
```
Comment (whyman) [inline src/content/scripting/script.cc:806]: Doesnt this change the behaviour of the method?

File: src/database/mysql/mysql_database.cc [hunk 2]
```diff
@@ -284,6 +284,21 @@ std::string MySQLDatabase::quote(std::string value) const
     return ret;
 }
 
+std::string MySQLDatabase::quote(std::string_view value) const
```
Comment (whyman) [inline src/database/mysql/mysql_database.cc:287]: Wouldn't a generic method that takes a char array and length be better? this is pretty much a copy of the above?

**Comment by whyman** [comment]
You cant assume that `data()` gives you a null terminated string, it often wont. This feels dodgy to me,

File: src/content/scripting/script.cc (hunk 1 above)
Comment (neheb) [reply src/content/scripting/script.cc:806]: No. Initially I moved the default section out of the switch but then got an unhandled switch statement warning. Note how the return how matches the case name.

**Comment by neheb** [comment]
Sure I can. All these functions return a static null terminated string.

File: src/database/mysql/mysql_database.cc (hunk 2 above)
Comment (KarlStraussberger) [reply src/database/mysql/mysql_database.cc:287]: It should be possible to get rid of the above method complete (with a bit of work on the calls

File: src/database/sqlite3/sqlite_database.cc [hunk 3]
```diff
@@ -284,6 +284,14 @@ void Sqlite3Database::_exec(const char* query)
...
 }
 
+std::string Sqlite3Database::quote(std::string_view value) const
+{
+    char* q = sqlite3_mprintf("'%q'", value.data());
+    std::string ret = q;
+    sqlite3_free(q);
+    return ret;
+}
+
```
Comment (KarlStraussberger) [inline src/database/sqlite3/sqlite_database.cc:294]: Please try to get rid of the original method

Comment (neheb) [reply src/database/sqlite3/sqlite_database.cc]: I think this is what you meant.

**Pull request merged**